qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,396 @@
+            from qiskit import Aer
+            from qiskit.algorithms.optimizers import UMDA
+            from qiskit.algorithms import QAOA
+            from qiskit.utils import QuantumInstance
+
+
+            H2_op = (-1.052373245772859 * I ^ I) + \
+            (0.39793742484318045 * I ^ Z) + \
+            (-0.39793742484318045 * Z ^ I) + \
+            (-0.01128010425623538 * Z ^ Z) + \
+            (0.18093119978423156 * X ^ X)
+
+            p = 2  # Toy example: 2 layers with 2 parameters in each layer: 4 variables
+
+            opt = UMDA(maxiter=100, size_gen=20, n_variables=p*2, disp=True)
+
+            backend = Aer.get_backend('statevector_simulator')
+            vqe = QAOA(opt,
+                       quantum_instance=QuantumInstance(backend=backend),
+                       reps=p)
+
+            result = vqe.compute_minimum_eigenvalue(operator=H2_op)
+
+        If it is desired to modify the percentage of individuals considered to update the
+        probabilistic model, then this code can be used. Here for example we set the 60% instead
+        of the 50% predefined.
+
+        .. code-block:: python
+
+            opt = UMDA(maxiter=100, size_gen=20, n_variables=p*2, alpha = 0.6, disp=True)
+
+            backend = Aer.get_backend('statevector_simulator')
+            vqe = QAOA(opt,
+                       quantum_instance=QuantumInstance(backend=backend),
+                       reps=p)
+
+            result = vqe.compute_minimum_eigenvalue(operator=qubit_op)
+
+
+    References:
+
+        [1]: Vicente P. Soloviev, Pedro Larrañaga and Concha Bielza (2022, July). Quantum Parametric
+        Circuit Optimization with Estimation of Distribution Algorithms. In 2022 The Genetic and
+        Evolutionary Computation Conference (GECCO). DOI: https://doi.org/10.1145/3520304.3533963
+
+        [2]: Vicente P. Soloviev. Python package EDAspy.
+        https://github.com/VicentePerezSoloviev/EDAspy.
+    """
+
+    ELITE_FACTOR = 0.4
+    STD_BOUND = 0.3
+
+    # setting = "---"
+
+    def __init__(
+        self, maxiter: int, size_gen: int, n_variables: int, alpha: float = 0.5, disp: bool = False
+    ) -> None:
+        r"""
+        Args:
+            maxiter: Maximum number of function evaluations.
+            size_gen: Population size of each generation.
+            n_variables: Number of variables to be optimized. For example in QAOA the number of variables
+                is 2p, where p is the number of layers of the circuit.
+            alpha: Percentage [0, 1] of the population to be selected as elite selection.
+            disp: Set to True to print convergence messages.
+        """
+
+        self.__disp = disp
+        self.__size_gen = size_gen
+        self.__max_iter = maxiter
+        assert 0 < alpha < 1, (
+            "Alpha represents a percentage and should be greater than 0 but lower " "than 1"
+        )
+        self.__alpha = alpha
+        self.__n_variables = n_variables
```

Comment:
Can we not infer this from the size of the initial point in minimize i.e. `x0`. This may not be known upfront. At least I assume the len of x0 is the same. We have tried to separate problem from configuration so that say a VQE plus optimizer of users choice can be created upfront and later will be used to solve problem they have. When it gets to the point of solving, then the number of variable etc would be known.

Reply:
That is true. Now I infer this value from the initial_point. Thus, the getter and setter of n_variables have been removed and this has been updated in the examples for the documentation. This makes the algorithm call much easier for users. Thanks.